core/controller/src/main/scala/whisk/core/controller/RestAPIs.scala

```diff
@@ -160,7 +160,7 @@ protected[controller] class RestAPIVersion(apipath: String, apiversion: String)(
             sendCorsHeaders {
                 (pathEndOrSingleSlash & get) {
                     complete(OK, info)
-                } ~ authenticate(basicauth) {
+                } ~ (authenticate(basicauth) | authenticate(certificateAuth)) {
```

Comment:
* since the certificate based authentication is deployment dependent, perhaps this matching should be guarded by a configuration parameter
answer: actully, the certificate desn't depend on the deployment process, the client certificate generated process in deployment is just an examle like server certificate. as below codes
```
  - name: gen untrusted client certificate for host
    local_action: shell "{{ playbook_dir }}/roles/nginx/files/genssl.sh" "*.{{ whisk_api_localhost_name | default(whisk_api_host_name) | default(whisk_api_localhost_name_default) }}" "client"
    when: nginx.ssl.client_ca_cert == "openwhisk-client-ca-cert.pem"
```
you can see, if <nginx.ssl.client_ca_cert> doesn't equal with `openwhisk-client-ca-cert.pem`, the untrusted client certificate genderated ansible task will not be executed. so in users's own env, they just set <nginx.ssl.client_ca_cert> another name, can use their own client certificate.

* is it possible to use client authentication without enabling the flag you added in the previous pull request?
answer: it is better to use the flag <nginx.ssl.verify_client>.